akka-http-core/src/main/scala/akka/http/scaladsl/ConnectionContext.scala

```diff
@@ -28,18 +30,31 @@ object ConnectionContext {
     enabledCipherSuites: Option[immutable.Seq[String]] = None,
     enabledProtocols:    Option[immutable.Seq[String]] = None,
     clientAuth:          Option[TLSClientAuth]         = None,
-    sslParameters:       Option[SSLParameters]         = None) =
-    new HttpsConnectionContext(sslContext, sslConfig, enabledCipherSuites, enabledProtocols, clientAuth, sslParameters)
+    sslParameters:       Option[SSLParameters]         = None,
+    http2:               UseHttp2                      = UseHttp2.Negotiated) =
+    new HttpsConnectionContext(sslContext, sslConfig, enabledCipherSuites, enabledProtocols, clientAuth, sslParameters, http2)
   //#https-context-creation
 
+  // for binary-compatibility, since 1.10.2
+  @deprecated
```

Comment:
deprecated should have since and reason filled in, should also have `@Deprecated`

Reply:
Didn't we say there is no need for the Java deprecation over in the Akka repo?

Reply:
I figured as it's in the `scaladsl` package (and the javadsl doesn't inherit from the scaladsl) it might be fine here?